client/app/bundles/course/assessment/submission/reducers/liveFeedback.js

```diff
@@ -1,125 +1,172 @@
+  draft.feedbackByQuestion[questionId] = {
+    ...getOrDefault(draft.feedbackByQuestion, questionId, {}),
+    ...newFeedbacks,
+  };
+  saveStateToLocalStorage(draft);
+};
+
+const liveFeedbackReducer = function (state = initialState, action) {
   switch (action.type) {
-    case actions.LIVE_FEEDBACK_INITIAL: {
-      const { questionId } = action.payload;
-      return produce(state, (draft) => {
-        if (!(questionId in draft)) {
-          draft.feedbackByQuestion[questionId] = {
-            isRequestingLiveFeedback: true,
-            pendingFeedbackToken: null,
-            answerId: null,
-            liveFeedbackId: null,
-            feedbackFiles: {},
-          };
-        } else {
-          draft.feedbackByQuestion[questionId] = {
-            ...draft.feedbackByQuestion[questionId],
-            isRequestingLiveFeedback: true,
-          };
-        }
-      });
-    }
     case actions.LIVE_FEEDBACK_REQUEST: {
       const { token, questionId, liveFeedbackId, feedbackUrl } = action.payload;
       return produce(state, (draft) => {
         draft.feedbackUrl ??= feedbackUrl;
-        if (!(questionId in draft)) {
-          draft.feedbackByQuestion[questionId] = {
-            isRequestingLiveFeedback: false,
-            liveFeedbackId,
-            pendingFeedbackToken: token,
-          };
-        } else {
-          draft.feedbackByQuestion[questionId] = {
-            isRequestingLiveFeedback: false,
-            ...draft.feedbackByQuestion[questionId],
-            liveFeedbackId,
-            pendingFeedbackToken: token,
-          };
-        }
+        const conversation = getConversation(draft, questionId);
+        updateFeedbackForQuestion(draft, questionId, {
+          isRequestingLiveFeedback: true,
+          liveFeedbackId,
+          pendingFeedbackToken: token,
+          conversation,
+          suggestedReplies: [],
+          isDialogOpen: true,
+        });
       });
     }
     case actions.LIVE_FEEDBACK_SUCCESS: {
       const { questionId, answerId, feedbackFiles } = action.payload;
+      const newFeedbacks = feedbackFiles.reduce(
+        (feedbackArr, feedbackFile) => [
+          ...feedbackArr,
+          ...(feedbackFiles.length > 1
+            ? [
+                {
+                  text: `Filename: ${feedbackFile.path}`,
+                  sender: 'Codaveri',
+                  isBold: true,
+                },
+              ]
+            : []),
+          ...feedbackFile.feedbackLines
+            .sort((a, b) => a.linenum - b.linenum)
+            .map((line, index) => ({
+              text: `Line ${line.linenum}: ${line.feedback}`,
+              sender: 'Codaveri',
+              linenum: line.linenum,
+              timestamp:
+                index === feedbackFile.feedbackLines.length - 1
+                  ? new Date().toISOString()
+                  : null,
+              isBold: false,
+            })),
```

Comment:
Ben also requested to collapse feedback which share the same line, please include that